package.json

```diff
@@ -4,8 +4,55 @@
   "description": "A Query Language and Runtime which can target any service.",
   "license": "MIT",
   "private": true,
-  "main": "index",
+  "main": "index.js",
   "module": "index.mjs",
+  "exports": {
+    ".": {
+      "types": {
+        "import": "./index.js.d.mts",
+        "default": "./index.d.ts"
+      },
+      "module": "./index.mjs",
+      "import": "./index.js.mjs",
+      "default": "./index.js"
+    },
```

Comment:
Beautiful, isn't it? 😬 
At least like this we avoid renaming any existing files.